core/arch/opnd.h

```diff
@@ -2760,12 +2760,32 @@ DR_API
 /**
  * Sets the register \p reg in the passed in mcontext \p mc to \p value.
  * \p mc->flags must include DR_MC_CONTROL and DR_MC_INTEGER.
- * \note Current release is limited to setting pointer-sized registers only
+ * \note This function is limited to setting pointer-sized registers only
  * (no sub-registers, and no non-general-purpose registers).
+ * See \p reg_set_value_ex for setting other register values.
  */
 void
 reg_set_value(reg_id_t reg, dr_mcontext_t *mc, reg_t value);
 
+DR_API
+/**
+ * Sets the register \p reg in the passed in mcontext \p mc to the value
+ * stored in the buffer \p val_buf.
+ *
+ * \p mc->flags must include DR_MC_CONTROL and DR_MC_INTEGER.
+ *
+ * Unlike \p reg_set_value, this function supports not only general purpose
+ * registers, but SIMD registers too. Does not yet support MMX registers.
+ *
+ * Up to sizeof(dr_zmm_t) bytes will be read from \p val_buf. The \p size
+ * paramter indicates the size of the input buffer.
+ *
+ * Returns false if the size is invalid, e.g., the register is smaller
+ * than the size of the input buffer.
+ */
+bool
+reg_set_value_ex(reg_id_t reg, dr_mcontext_t *mc, IN byte *val_buf, IN size_t size);
```

Comment:
Add a comment at appropriate place in api/docs/release.dox (non-compatibility change)

Reply:
Done